src/services/upload_terminal.js

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright: Ambrosus Technologies GmbH
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import {getDefaultAddress} from '../utils/web3_tools';
+
+
+export default class UploadTerminal {
```

Comment:
I would replace UploadTerminal with 2 repositories (for fees and uploads)

Reply:
I'd leave it like this. We would have about 10 different `ContractRepository` classes.
I would rather see them responsible for whole piece of work, instead of particular contract they interact with.

Reply:
I'm with @Anze1m on this. The split into different contracts is a implementation detail of our smart contracts which are a service layer for the node. The interface (called from the model engine) should be very simple here: `upload this stuff, I don't care about the details.` And hence a single provider of this interface is a valid idea.